packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,268 @@
+   * @private
+   */
+  private _trigger?: HTMLElement;
+
+  /**
+   * Container that holds the menu list items.
+   * @public
+   */
+  public menuListContainer?: HTMLElement;
+
+  /**
+   * Holds a reference to a function that is used to cleanup resources.
+   * @private
+   */
+  private cleanup?: () => void;
+
+  /**
+   * Method to toggle the expanded state of the menu on an event.
+   * @param {MouseEvent} e - The event that triggered the expansion of the menu.
+   * @public
+   */
+  public toggleMenu = () => {
+    this.expanded = !this.expanded;
+  };
+
+  /**
+   * Method to close the menu.
+   * @public
+   */
+  public closeMenu() {
+    this.expanded = false;
+  }
+
+  /**
+   * Method to open the menu.
+   * @public
+   */
+  public openMenu() {
+    this.expanded = true;
+  }
+
+  /**
+   * Method to focus on the menu.
+   * @public
+   */
+  public focus(): void {
+    if (this.expanded && this._menu) {
+      const menu = this._menu as MenuList;
+      Updates.enqueue(() => menu.focus());
+    }
+  }
+
+  /**
+   * Method to set the positioning of the menu.
+   * @public
+   */
+  public setPositioning(): void {
+    if (this.$fastController.isConnected && this._trigger && this.menuListContainer) {
+      this.cleanup = autoUpdate(this, this.menuListContainer, async () => {
+        const { middlewareData, x, y } = await computePosition(this._trigger!, this.menuListContainer!, {
+          placement: 'bottom',
+          strategy: 'fixed',
+          middleware: [
+            flip(),
+            size({
+              apply: ({ availableHeight, rects }) => {
+                this.menuListContainer?.style &&
+                  Object.assign(this.menuListContainer.style, {
+                    maxHeight: `${availableHeight}px`,
+                    width: `${rects.reference.width}px`,
+                  });
+              },
+            }),
+            hide(),
+          ],
+        });
+        if (middlewareData.hide?.referenceHidden) {
+          this.expanded = false;
+          return;
+        }
+
+        this.menuListContainer?.style &&
+          Object.assign(this.menuListContainer.style, {
+            position: 'fixed',
+            top: '0',
+            left: '0',
+            transform: `translate(${x}px, ${y}px)`,
+          });
+      });
+    }
+  }
+
+  /**
+   * Method that gets called whenever the menu changes.
+   * @protected
+   */
+  protected menuChanged(): void {
+    if (this.$fastController.isConnected && this.menu && this.menu.length) {
+      this._menu = this.menu[0] as MenuList;
+    }
+  }
+
+  /**
+   * Method that gets called whenever the trigger changes.
+   * @protected
+   * @param {HTMLElement[] | void} prev - The previous value of the trigger.
+   * @param {HTMLElement[]} next - The new value of the trigger.
+   */
+  protected triggerChanged(prev: HTMLElement[] | void, next: HTMLElement[]): void {
+    if (prev !== next) {
```

Comment:
this only gets invoked if `prev !== next`, if these are the same the change handler won't fire, I think this check is redundant.